backend/django/core/forms.py

```diff
@@ -418,3 +457,71 @@ def clean_data(self):
             return cleanCodebookDataHelper(data)
         else:
             return ""
+
+
+class ExternalDatabaseWizardForm(forms.ModelForm):
+    class Meta:
+        model = ExternalDatabase
+        fields = ["database_type", "ingest_schema", "ingest_table_name"]
+
+    database_type = forms.ChoiceField(
+        widget=RadioSelect(),
+        choices=ExternalDatabase.DB_TYPE_CHOICES,
+        initial="none",
+        required=True,
+    )
+    ingest_table_name = forms.CharField(initial="", required=False, max_length=50)
+    ingest_schema = forms.CharField(initial="", required=False, max_length=50)
+    username = forms.CharField(initial="", required=False, max_length=50)
+    password = forms.CharField(
+        initial="", required=False, max_length=200, widget=forms.PasswordInput()
+    )
+    host = forms.CharField(initial="", required=False, max_length=50)
+    port = forms.IntegerField(required=False)
+    dbname = forms.CharField(initial="", required=False, max_length=50)
+    driver = forms.CharField(
+        initial="ODBC Driver 17 for SQL Server", required=False, max_length=200
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(ExternalDatabaseWizardForm, self).__init__(*args, **kwargs)
+
+    def clean(self):
+        required_for_all_db = [
+            "ingest_table_name",
+            "ingest_schema",
+            "username",
+            "password",
+            "host",
+            "port",
+            "dbname",
+        ]
+        required_for_ms_sql = ["driver"]
+        db_type = self.cleaned_data.get("database_type")
+
+        field_error = False
+        if db_type != "none":
+            for field in required_for_all_db:
+                if self.cleaned_data.get(field) in ["", None]:
+                    self._errors[field] = self.error_class(
+                        ["This field is required for a database connection."]
+                    )
+                    field_error = True
+            if db_type == "microsoft":
+                for field in required_for_ms_sql:
+                    if self.cleaned_data.get(field) in ["", None]:
+                        self._errors[field] = self.error_class(
+                            ["This field is required for a MS SQL database connection."]
+                        )
+                        field_error = True
+            if field_error:
+                raise ValidationError("Please fix field errors before resubmitting.")
+
+            engine_database = get_connection(db_type, self.cleaned_data)
+            test_connection(
```

Comment:
I also like the test of the database connection and think it is helpful. I made a typo when I was first inputting the connection information and the error helped me figure it out